src/os/ObjectStore.h

```diff
@@ -348,6 +348,7 @@ class ObjectStore {
       OP_COLL_SET_BITS = 42, // cid, bits
 
       OP_MERGE_COLLECTION = 43, // cid, destination
+      OP_OMAP_SETTAIL = 44, // cid, tail
```

Comment:
I don't think this should be exposed through the ObjectStore interface.  Instead, we should create the head and tail keys whenever we set the FLAG_OMAP, as part of this block:
```
  if (!o->onode.has_omap()) {
    o->onode.set_omap_flag();
    if (o->oid.is_pgmeta()) {
      o->onode.flags |= bluestore_onode_t::FLAG_PGMETA_OMAP;
    }
    txc->write_onode(o);
  } else {
```
which appears in setkeys, setheader, and clone.

Reply:
we should also make sure ``do_omap_clear()`` removes the tail record too

Reply:
ok. fixed